design/nab_status_update_ver_1.md

```diff
@@ -0,0 +1,86 @@
+| Running | Velero *Backup* is currently running |
+| Succeeded | Velero *Backup* was performed with Success state |
+| Failed | Velero *Backup* did not succeed |
+
+### Conditions
+
+The `conditions` is also a part of the NonAdminBackup's `status` field. One NAB object may have multiple conditions. It is more granular knowledge of the NonAdminBackup object and represents the array of the conditions through which the NonAdminBackup has or has not passed. Each `NonAdminCondition` has one of the following `type`:
+
+| **Condition** | **Description**                 |
+|-----------|--------------------------------|
+| BackupAccepted | The Backup object was accepted by the reconcile loop, but the Velero Backup may have not yet been created |
+| BackupQueued | The Velero Backup was created succesfully and did not return any known errors. It's in the queue for Backup. At this stage errors may still occur during backup procedure. |
+| BackupCompleted | The Velero Backup succeeded. |
+
+The `condition` data is also accomapied with the following:
+
+| **Field name** | **Description**                 |
+|-----------|--------------------------------|
+| type | The `Type` of the condition |
+| status | represents the state of individual condition. The resulting `phase` value should report `Success` only when all of the conditions are met and the backup succeded. One of `True`, `False` or `Unknown`. |
+| lastProbeTime | Timestamp of when the NonAdminBackup condition was last probed. |
+| lastTransitionTime | Timestamp for when the NonAdminBackup last transitioned from one status to another. |
+| reason | Machine-readable, UpperCamelCase text indicating the reason for the condition's last transition. |
+| message | Human-readable message indicating details about the last status transition. |
+
+### BackupStatus
+
+`BackupStatus` which is also part of the `NonAdminBackupStatus` object is a `BackupStatus` that is taken directly from the Velero Backup Status and copied over.
+
+## Phase Update scenarios
+
+    *** Questions ***
+
+     - BackupQueued stays true at the end,
+       should we remove that from conditions?
+       I don't think we should set it to false.
+
+```mermaid
+graph
+AA[Phase: New] -->A
+A{BackupAccepted: Unknown\n BackupQueued: False\n BackupCompleted: False}  -- NAC processes --> B[Non Admin Backup Accepted]
+A -- NAC hits an error --> E[Phase: BackingOff]
+B -- Create Velero Backup --> C[Phase: Created]
+B -.-> BA{BackupAccepted: True\nBackupQueued: False\nBackupCompleted: False}
+C  -- Backup runs --> D[Phase: Running]
+C -.-> BS{BackupAccepted: True\nBackupQueued: True\nBackupCompleted: False}
+D -- Backup succeeds --> F[Phase: Succeeded]
+F -.-> FF{BackupAccepted: True\nBackupQueued: True\nBackupCompleted: True}
+
+D -- Backup fails --> G[Phase: Failed]
+E -.-> EE{BackupAccepted: False\nBackupQueued: False\nBackupCompleted: False}
+
+classDef conditions fill:#ccc,stroke:#ccc,stroke-width:2px;
```

Comment:
never used mermaid, can we change colors? for me in Dark Mode, I do not see what is written inside these blocks
![Screenshot from 2024-04-03 11-55-29](https://github.com/migtools/oadp-non-admin/assets/66965232/e2713121-e142-4ed2-bda4-d020827aa4b4)

Reply:
Should be resolved with https://github.com/migtools/oadp-non-admin/pull/23/files#r1550044291

tested at https://mermaid.live/edit#pako:eNqlk0uL2zAUhf-KUAnZKF10E1Cg4Pgxq6aPtN2MsxDS9VjElowkkxbj_17JitpMZwqBghfiSud891x8J8y1AEzxajVJJR1F09q10MOarhttwLr1PK9WtXoybGhrlWWPn1pmgaIDXE5os3mf-eK0Z_w8DhnnMDgQFH1TZ6Uvqq4VilefRxjDRcU6C3_Kue6HDtzvmxl5S3TIcjQYzcFasIGB9o8HrVAmepmUKMFOnp9ErXQWMYXAGG0WYZnaDSqpnj42jRfsgyA3wByg79CB0ck1aPKkiS9EFLwNbbxM-tWMIc_rKf8Rslb5kvPKNKOKKYsE_jIq5Zs9hYcRfLwP_Kz4Cre4wdrRe4GI6Cqhj7G6pK4ivKr-Dx7qf7EbJrsIfkjgypcWahmpZfmC-nysdw-bd8zaAhrEtRLSSe3H3ciuo28458Q6o89we95cpHAtfTf82F3FKCN7_x3JPidlSarqxmp3AxhClGS-3W7vNM9ITgpSkgdSXS12mOAeTM-k8Ls5-X1BNV72ssbUHwUz5xrXavbv2Oj08afimDo_aILHQfi_tpDMr2yPaROmQDD4drX5EJd92fn5F-f7XJ0

Reply:
applied @kaovilai changes in vscode and got a good result! adding theme and removing hardcoded "CSS" should resolve for dark mode users